homeassistant/components/reolink/binary_sensor.py

```diff
@@ -121,6 +137,140 @@ class ReolinkBinarySensorEntityDescription(
     ),
 )
 
+BINARY_SMART_SENSORS = (
+    ReolinkSmartBinarySensorEntityDescription(
+        key="crossline",
+        ai_type="people",
+        cmd_id=33,
+        translation_key="crossline_person",
+        value=lambda api, ch, loc: (
```

Comment:
instead of duplicating the definition of `value`, we could add `smart_type: str` to `ReolinkSmartBinarySensorEntityDescription`, and implement `value` as: 
```python
    def value(self, api: Host, ch: int, loc: int) -> bool:
        return api.baichuan.smart_ai_state(ch, self.smart_type, loc, self.ai_type)
```

We could probably also do it for `supported()`, so the actual description of a sensor is simplified to:
```python
    ReolinkSmartBinarySensorEntityDescription(
        key="crossline",
        ai_type="people",
        smart_type="crossline",
        cmd_id=33,
        translation_key="crossline_person",
    ),
```

Reply:
Or maybe better to just remove `value` completely from the Description and call `api.baichuan.smart_ai_state(ch, description.smart_type, loc, description.ai_type)` in the `ReolinkSmartBinarySensorEntity.is_on`

Reply:
Yea I was thinking about this.
However with the current approach, it is setup more general such that I can add more binary sensors in the future withouth needing to touch the class.

But I do agree it could be a lot more compact if I give up on that generalization.

It is now more in line with the other binary sensors though.

Reply:
If it is the same to you, I would like to keep it this way to keep it uniform accross all platforms.
This will also help if Reolink (or I) decides to add stuff which would fit a binary smart AI sensor.